keras/layers/rnn/bidirectional_test.py

```diff
@@ -1020,6 +1020,49 @@ def test_full_input_spec(self):
         self.assertAllClose(output1, output3)
         self.assertNotAllClose(output1, output2)
 
+    def test_reset_states(self):
+        ref_state = np.random.rand(1, 3).astype(np.float32)
+
+        # build model
+        inp = keras.Input(batch_shape=[1, 2, 3])
+
+        stateful = keras.layers.SimpleRNN(units=3, stateful=True)
+        stateless = keras.layers.SimpleRNN(units=3, stateful=False)
+
+        bid_stateless = keras.layers.Bidirectional(stateless)
+        bid_stateful = keras.layers.Bidirectional(stateful)
+
+        _ = keras.Model(
```

Comment:
Any reason to build a model and doesn't use it?

Reply:
This is required to correctly initialize the state within the layers. Otherwise calling `reset_state` on any of them causes a `ValueError`.